### PR TITLE
Add TLS and STARTTLS support.

### LINKED ISSUE #31: Add simple smtps as recommended by RFC 8314 for mail submission of endusers
The heading says it all.

Happy to provide a pr.

### PR DESCRIPTION
**Pull request by typetetris**
Should close #31

### CODE REVIEW

**Comment by parsonsmatt** [comment]
Would you like to  be a maintainer on the project? I'm not really using the library anymore and it would be great to have more active eyes/ears on it.

@jhickner thoughts?

**Comment by jhickner** [comment]
Yeah, sounds good to me @parsonsmatt.

**Comment by typetetris** [comment]
I thought about this some and the library is small an I will use it for some time, so I would try my best being a maintainer for it. Would you also include me in the hackage maintainer group, so I could do uploads? My hackage account has username typetetris too.

**Comment by jhickner** [comment]
Ok I added github access for you, and I'm looking into how to add you to hackage.

**Comment by typetetris** [comment]
On http://hackage.haskell.org/package/smtp-mail/maintainers/ it should be possible to add me.

**Comment by jhickner** [comment]
Ok, done

**Comment by typetetris** [comment]
@jhickner @parsonsmatt Would you review my changes please?
If you don't have the time for that, please tell me, then I will just proceed on my own.

**Comment by parsonsmatt** [comment]
I'm on vacation this week and won't have time. Happy to review after, though you may want to ping me again next weekend :)

**Comment by piq9117** [comment]
thanks for working on this @typetetris I tried this out today with gmail smtp. it works great!

**Comment by typetetris** [comment]
> I'm on vacation this week and won't have time. Happy to review after, though you may want to ping me again next weekend :)

Ping :smile:

**Comment by typetetris** [comment]
@parsonsmatt Another ping. If you don't find the time, just let me know.

**Comment by parsonsmatt** [comment]
Hey, sorry, weekends aren't a great time to ping me - I tend to be away from the computer a lot then. I'll try to get to this today.

**Comment by typetetris** [comment]
No worries. Knowing that, I will ping in the future on workdays.

**Comment by typetetris** [comment]
So, hopefully your not too annoyed, but here is another @parsonsmatt ping

File: README.md [hunk 1]
```diff
@@ -42,6 +42,18 @@ or, with authentication:
...
+
+```haskell
+main = sendMailSTARTTLS host mail
+```
+
+or, using SMTPS:
+
+```haskell
+main = sendMailTLS host mail
+```
```
Comment (parsonsmatt) [inline README.md:55]: docs <3 thanks!

File: README.md [hunk 2]
```diff
@@ -62,12 +74,19 @@ For more complicated scenarios or for adding attachments or CC/BCC
...
 make common operations easier, and switched to ByteStrings where applicable.
+
+### Developing
+
+`nix-integration-test/integration-test.nix` contains a integration test, which
+uses nixos qemu vm tests to start a qemu vm with a postfix and use smtp-mail to
+send mails to that postfix.
+
+Install [nix](https://nixos.org) and execute `nix-build nix-integration-test/integration-test.nix`
+to execute the test. Success is signalled by a return code of `0`.
```
Comment (parsonsmatt) [inline README.md:90]: Very cool!

**Pull request merged**